core/DataAccess/LogAggregator.php

```diff
@@ -1100,6 +1100,129 @@ public function queryConversionsByDimension($dimensions = array(), $where = fals
         return $this->getDb()->query($query['sql'], $query['bind']);
     }
 
+    /**
+     * Similar to queryConversionsByDimension and will return data in the same
+     * format, but takes into account pageviews leading up to a conversion, not
+     * just the final page that triggered the conversion
+     *
+     * @param string $linkField
+     *
+     * @return \Zend_Db_Statement|array
+     */
+    public function queryConversionsByPageView(string $linkField)
+    {
+        $dbSettings = new \Piwik\Db\Settings();
+        $tablePrefix = $dbSettings->getTablePrefix();
+        $subQuery = sprintf("
+            MAX((SELECT COUNT(am.idaction)
+             FROM %slog_conversion cam
+             LEFT JOIN %slog_link_visit_action vam ON vam.idvisit = cam.idvisit AND vam.server_time <= log_conversion.server_time
```

Comment:
The query still throws the same error for me. Does this maybe need to be
```suggestion
             LEFT JOIN %slog_link_visit_action vam ON vam.idvisit = cam.idvisit AND vam.server_time <= cam.server_time
```

Reply:
@sgiehl The log_link_visit_action.server_time from the sub-query (vam) is meant to join against the outer query log_conversion.server_time so `vam.server_time <= log_conversion.server_time` is intentional. This is working perfectly for me both on MariaDB and MySQL, so there must be some other difference in database setup. Are you using table prefixes? PDO or MySQLi? Which database version? I can try to match your test setup and recreate the issue.

Below is the extracted query in full, does it work for you if run directly? (it uses the table prefix 'matomo_' so will need a search and replace if you use something different)
```
SELECT
  log_conversion.idgoal AS idgoal,
  log_link_visit_action.idaction_url AS idaction,
  COUNT(*) AS `10`,
  COUNT(distinct log_conversion.idvisit) AS `3`,
  ROUND(SUM(log_conversion.revenue),2) AS `2`,
  ROUND(SUM(log_conversion.revenue_subtotal),2) AS `4`,
  ROUND(SUM(log_conversion.revenue_tax),2) AS `5`,
  ROUND(SUM(log_conversion.revenue_shipping),2) AS `6`,
  ROUND(SUM(log_conversion.revenue_discount),2) AS `7`,
  SUM(log_conversion.items) AS `8`,
  MAX((SELECT COUNT(am.idaction)
   FROM matomo_log_conversion cam
   LEFT JOIN matomo_log_link_visit_action vam ON vam.idvisit = cam.idvisit AND vam.server_time <= log_conversion.server_time
   LEFT JOIN matomo_log_action am ON am.idaction = vam.idaction_url
   WHERE cam.idgoal = log_conversion.idgoal AND vam.idvisit = log_link_visit_action.idvisit
   AND vam.idaction_url IS NOT NULL AND am.type = 1
   GROUP BY cam.idgoal, cam.idvisit
  )) AS `9`,
  1 AS `12`
FROM matomo_log_visit AS mv
INNER JOIN matomo_log_conversion AS log_conversion ON log_conversion.idvisit = mv.idvisit
LEFT JOIN matomo_log_link_visit_action AS log_link_visit_action ON log_link_visit_action.idvisit = log_conversion.idvisit
                                                                   AND log_link_visit_action.server_time <= log_conversion.server_time
LEFT JOIN matomo_log_action AS log_action ON log_action.idaction = log_link_visit_action.idaction_url
WHERE log_conversion.server_time >= '2021-11-03 11:00:00'
  AND log_conversion.server_time <= '2021-11-04 10:59:59'
  AND log_conversion.idsite IN (1) AND log_conversion.idgoal > 0
  AND log_link_visit_action.idaction_url IS NOT NULL
  AND log_action.type = 1
GROUP BY
  log_conversion.idgoal, log_conversion.idvisit, log_action.idaction;
```

Reply:
@bx80 it fails on travis ci as well. My setup is similar. See https://app.travis-ci.com/github/matomo-org/matomo/jobs/549249575#L795

Reply:
This query doesn't work in AWS Aurora MySQL 5.7.12, so leave it with me and I can test against that.

Reply:
Ok, so the query worked with MariaDB 10.x and MySQL 8.x but not MySQL 5.7. Looks like some sort of issue with correlated subquery joins with more than one condition. I've adjusted the query to move the extra join condition to the where clause and the query now works on MySQL 5.7

Reply:
Fine. Had a quick look and archiving seems to run through with those changes. Will continue my code review next week then.